pkg/clockcache/cache_bench_test.go

```diff
@@ -236,6 +237,62 @@ func BenchmarkInsertConcurrent(b *testing.B) {
 	}
 }
 
+// BenchmarkCacheFalseSharing is a benchmark to measure the effect of the false
+// sharing of CPU cache lines. In the clockcache.element struct, we introduce
+// padding to ensure that only one clockcache.element fits in a CPU cache line,
+// avoiding false sharing.
+//
+// The principle behind this benchmark is simple: construct a cache with two
+// entries, and start two goroutines which each clobber one of the cache values
+// over and over again. If there is false sharing, it should be measurable by
+// toggling the padding on and off, and measuring the difference in output of
+// this benchmark.
+//
+// At the time of writing, this code was tested on an M1 MacBook Pro, where the
+// advantage obtained by introducing padding is approximately 16%:
+//     go test -bench=BenchmarkCacheFalseSharing -cpu=2 -count=10 > no-padding.txt
+//     go test -bench=BenchmarkCacheFalseSharing -cpu=2 -count=10 > padding.txt
+//     benchstat no-padding.txt padding.txt
+//       name                 old time/op    new time/op    delta
+//       CacheFalseSharing-2     230ns ± 6%     193ns ±19%  -16.09%  (p=0.001 n=10+9)
+//
+// Note: This benchmark _must_ be run with the `-cpu=2` argument, to ensure
+// that each goroutine ends up on a different CPU, possibly causing contention
+// for the same cache line.
+func BenchmarkCacheFalseSharing(b *testing.B) {
+	cache := WithMax(2)
+	b.ReportAllocs()
+
+	// define waitgroup, so we can wait until background incrementing tasks are finished
+	wg := &sync.WaitGroup{}
+	wg.Add(2)
+
+	key1 := 0
+	key2 := 1
+	times := b.N
+	b.ResetTimer()
```

Comment:
Nit: Could we reset timer in the stressor? That way we ensure the last goroutine to start will reset the timer and just measure after the cache trashing has started.

Reply:
I guess we could, but I suspect that could end up distorting the results too. I've done what you suggested, but with an additional waitgroup to coordinate the goroutine start.